internal/objects/service/service.go

```diff
@@ -260,6 +260,9 @@ func DesiredEnvoyService(contour *operatorv1alpha1.Contour) *corev1.Service {
 	switch epType {
 	case operatorv1alpha1.LoadBalancerServicePublishingType:
 		svc.Spec.Type = corev1.ServiceTypeLoadBalancer
+		if len(contour.Spec.NetworkPublishing.Envoy.LoadBalancer.ProviderParameters.LoadBalancerIP) > 0 {
```

Comment:
would be good to have a unit test case for this

Reply:
i think that should be sufficient for now, its hard to do an end to end test for this b/c of kind + LoadBalancerService